nemo/src/model/types/primitive_logical_value.rs

```diff
@@ -341,33 +343,126 @@ impl TryFrom<Term> for PhysicalString {
+                            if !remainder.is_empty() {
+                                Err(InvalidRuleTermConversion::new(
+                                    term.clone(),
+                                    PrimitiveType::Any,
+                                ))
+                            } else {
+                                match number {
+                                    NumericLiteral::Integer(i) => {
+                                        Ok(LogicalFloat64::try_from(LogicalInteger::from(i))
+                                            .map(|d| d.into())
+                                            .unwrap_or(
+                                                DatatypeValue(
+                                                    value.to_string(),
+                                                    datatype.to_string(),
+                                                )
+                                                .into(),
+                                            ))
+                                    }
+                                    NumericLiteral::Decimal(a, b) => Ok(format!("{a}.{b}")
+                                        .parse()
+                                        .ok()
+                                        .and_then(|d| {
+                                            Double::new(d)
+                                                .map(|d| LogicalFloat64::from(d).into())
+                                                .ok()
+                                        })
+                                        .unwrap_or(
+                                            DatatypeValue(value.to_string(), datatype.to_string())
+                                                .into(),
+                                        )),
+                                    NumericLiteral::Double(d) => Ok(LogicalFloat64::from(d).into()),
+                                }
+                            }
+                        }
+                        Err(nom::Err::Error(located_parse_error)) => {
+                            let parse_error: &ParseError =
+                                located_parse_error.get_last_inner().into();
+
+                            match parse_error {
+                                ParseError::SyntaxError(_) => Err(InvalidRuleTermConversion::new(
+                                    term.clone(),
+                                    PrimitiveType::Any,
+                                )),
+                                _ => {
+                                    Ok(DatatypeValue(value.to_string(), datatype.to_string())
+                                        .into())
+                                }
+                            }
+                        }
+                        _ => Err(InvalidRuleTermConversion::new(
+                            term.clone(),
+                            PrimitiveType::Any,
+                        )),
                     }
```

Comment:
Note that `xsd:double` (and `xsd:float`) are distinct, finite datatypes with a value space disjoint from that of `xsd:decimal`. Not only is it perfectly fine to only accept what parses as `f64` (and `f32` for `xsd:float`, respectively), it's actually _required_.

Reply:
I see, that makes things simpler :)